app/src/main/java/ch/epfl/sweng/rps/ui/home/MatchViewModel.kt

```diff
@@ -20,51 +22,59 @@ import java.util.*
 class MatchViewModel : ViewModel() {
 
 
-    private var _gameService: OfflineGameService? = null
+    var gameService: OfflineGameService? = null
     var currentUserResult: Hand.Result? = null
+    var computerPlayer: ComputerPlayer? = null
     private val uid = FirebaseAuth.getInstance().currentUser?.uid
     fun startOfflineGameService(
         nEvents: Int,
         computerPlayer: ComputerPlayer,
         artificialMovesDelay: Long = 1_000
     ) {
         // TODO: when having a local DB change the random UUID
+        this.computerPlayer = computerPlayer
         val gameId = UUID.randomUUID().toString()
-        _gameService = OfflineGameService(
+        gameService = OfflineGameService(
             gameId,
             ServiceLocator.getInstance().repository,
             listOf(computerPlayer),
             Game.GameMode(2, Game.GameMode.Type.PC, nEvents, 0),
             artificialMovesDelay
         )
-        _gameService?.startListening()
+        gameService?.startListening()
     }
 
 
     private fun determineResult() {
-        val scores = _gameService?.currentRound!!.computeScores()
+        val scores = gameService?.currentRound!!.computeScores()
         val best = scores[0]
         val bestPoints = best.points
-        val bestUid = best.uid
-        currentUserResult = if (uid == bestUid) {
-            val secondBestPoints = scores[1].points
-            if (bestPoints > secondBestPoints) {
+        val bestScoreList = mutableListOf<Round.Score>()
+        for (score in  scores){
```

Comment:
Why not use a filter method to follow a more Kotlin-functional style?

Reply:
Thanks, I've changed it now.